lib/src/lifecycle_module.dart

```diff
@@ -321,11 +321,18 @@ abstract class LifecycleModule extends SimpleModule with Disposable {
     }
 
     _state = LifecycleState.loading;
-    _transition = new Completer<Null>();
 
-    _load().then(_transition.complete).catchError(_transition.completeError);
+    // Keep track of this load's completer
+    final transition = new Completer<Null>();
 
-    return _transition.future;
+    // because this one can get overwritten
+    _transition = transition;
+
+    _load().then(transition.complete).catchError((error, trace) {
+      transition.completeError(error, trace);
```

Comment:
Maybe there was something here before that necessitated the extra closure, I can't remember, but at this point we're just completing the transition with the result of `_load()`. I think you can go back to something like this:

```dart
_transition = new Completer<Null>();
_transition.complete(_load());
return _transition.future;
```

Since we're setting up the completion immediately, there's no need for the extra `transition` var – it doesn't matter if `_transition` gets overwritten later